src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -157,7 +180,30 @@ def _create_temporary_field(
     return Field(field_node, field_type)
 
 
-def translate_as_field_op(
+def translate_domain(node: gtir.Node) -> FieldopDomain:
+    """
+    Visits the domain of a field operator and returns a list of dimensions and
+    the corresponding lower and upper bounds.
+    """
+    assert cpm.is_call_to(node, ("cartesian_domain", "unstructured_domain"))
+
+    domain = []
+    for named_range in node.args:
+        assert cpm.is_call_to(named_range, "named_range")
+        assert len(named_range.args) == 3
+        axis = named_range.args[0]
+        assert isinstance(axis, gtir.AxisLiteral)
+        bounds = [
+            dace.symbolic.pystr_to_symbolic(gtir_python_codegen.get_source(arg))
+            for arg in named_range.args[1:3]
+        ]
+        dim = gtx_common.Dimension(axis.value, axis.kind)
+        domain.append((dim, bounds[0], bounds[1]))
+
+    return domain
+
+
+def translate_as_fieldop(
     node: gtir.Node,
```

Comment:
`FunCall` should be part of the type signature, but I would keep the assert.

Reply:
Then we keep it that way.
I do not see the point of it.

Reply:
The point is to ensure that all `translate_` functions have the same signature.